algolia/transport/transport.go

```diff
@@ -140,29 +144,56 @@ func (t *Transport) Request(
 	return errs.ErrNoMoreHostToTry
 }
 
+func isErrBecauseOfContextCancellation(ctx context.Context, err error) bool {
+	if err == nil {
+		return false
+	}
+
+	var (
+		isCancelled        = strings.Contains(err.Error(), context.Canceled.Error())
+		isDeadlineExceeded = strings.Contains(err.Error(), context.DeadlineExceeded.Error())
```

Comment:
Why do we have to check the `err` value? Wouldn't be enough to check for the context? Whatever the error we got we probably want to stop if the context is canceled no?

Reply:
Actually, I've forgot to take a step back from the original issue. Thanks for noticing. It makes sense to abort on any context cancellation, since the end goal is just to early abort the retry strategy when user-provided context is already cancelled.

Thanks for noticing!